sdk/android/src/java/org/webrtc/CameraCapturer.java

```diff
@@ -312,6 +312,20 @@ public void run() {
     Logging.d(TAG, "Stop capture done");
   }
 
+  @Override
+  public void setOrientation(int orientation) {
+    synchronized (stateLock) {
+      if (currentSession != null) {
```

Comment:
Don't we need to take the lock *inside* the Runnable below?  Otherwise, the lock is only used to post to the ThreadHandler, and then the Runnable doesn't have the lock.

Reply:
Yes, good catch, fixed.